feature/interface/aggregate/otg_tests/aggregate_all_not_viable_test/aggregate_all_not_forwarding_viable_test.go

```diff
@@ -0,0 +1,1025 @@
+	pfx2AdvV4                = &ipAddr{ip: "100.0.2.0", prefix: 24}
+	pfx2AdvV6                = &ipAddr{ip: "2003:db8:64:64::0", prefix: 64}
+	pfx3AdvV4                = &ipAddr{ip: "100.0.3.0", prefix: 24}
+	pfx4AdvV4                = &ipAddr{ip: "100.0.4.0", prefix: 24}
+	pmd100GFRPorts           []string
+	dutPortList              []*ondatra.Port
+	atePortList              []*ondatra.Port
+	rxPktsBeforeTraffic      map[*ondatra.Port]uint64
+	txPktsBeforeTraffic      map[*ondatra.Port]uint64
+	equalDistributionWeights = []uint64{50, 50}
+	ecmpTolerance            = uint64(1)
+)
+
+func TestMain(m *testing.M) {
+	fptest.RunTests(m)
+}
+
+// TestAggregateAllNotForwardingViable Test forwarding-viable with LAG and routing
+func TestAggregateAllNotForwardingViable(t *testing.T) {
+
+	dut := ondatra.DUT(t, "dut")
+	ate := ondatra.ATE(t, "ate")
+
+	aggIDs := configureDUT(t, dut)
+	changeMetric(t, dut, aggIDs[2], 30)
+	top := configureATE(t, ate)
+	installGRIBIRoutes(t, dut, ate, top)
+	flows := createFlows(t, top)
+	ate.OTG().PushConfig(t, top)
+	ate.OTG().StartProtocols(t)
+
+	for _, aggID := range aggIDs {
+		gnmi.Await(t, dut, gnmi.OC().Interface(aggID).OperStatus().State(), 30*time.Second, oc.Interface_OperStatus_UP)
+	}
+
+	for _, agg := range []*aggPortData{agg1, agg2, agg3} {
+		bgpPath := ocpath.Root().NetworkInstance(deviations.DefaultNetworkInstance(dut)).Protocol(oc.PolicyTypes_INSTALL_PROTOCOL_TYPE_BGP, "BGP").Bgp()
+		gnmi.Await(t, dut, bgpPath.Neighbor(agg.ateIPv4).SessionState().State(), time.Minute, oc.Bgp_Neighbor_SessionState_ESTABLISHED)
+	}
+
+	t.Logf("ISIS cost of LAG_2 lower then ISIS cost of LAG_3 Test-01")
+	t.Run("RT-5.7.1.1: Setting Forwarding-Viable to False on Lag2 all ports except port 2", func(t *testing.T) {
+		configForwardingViable(t, dut, dutPortList[2:agg2.ateLagCount+1], false)
+		startTraffic(t, dut, ate, top)
+		if err := checkBidirectionalTraffic(t, dut, dutPortList[1:2]); err != nil {
+			t.Fatal(err)
+		}
+		if err := confirmNonViableForwardingTraffic(t, dut, ate, atePortList[2:agg2.ateLagCount+1], dutPortList[2:agg2.ateLagCount+1]); err != nil {
+			t.Fatal(err)
+		}
+		// Ensure there is no traffic received/transmiited on DUT LAG_3
+		if got := validateLag3Traffic(t, dut, ate, dutPortList[(agg2.ateLagCount+1):]); got == true {
+			t.Fatal("Packets are Received and Transmitted on LAG_3")
+		}
+		if ok := verifyTrafficFlow(t, ate, flows); !ok {
+			t.Fatal("Packet Dropped, LossPct for flow ")
+		}
+	})
+	t.Run("RT-5.7.1.2: Setting Forwarding-Viable to False for Lag2 all ports", func(t *testing.T) {
+		// Ensure ISIS Adjacency is up on LAG_2
+		if ok := awaitAdjacency(t, dut, aggIDs[1], oc.Isis_IsisInterfaceAdjState_UP); !ok {
+			t.Fatal("ISIS Adjacency is Down on LAG_2")
+		}
+		configForwardingViable(t, dut, dutPortList[1:2], false)
+		// Ensure ISIS Adjacency is Down on LAG_2
+
+		if ok := awaitAdjacency(t, dut, aggIDs[1], oc.Isis_IsisInterfaceAdjState_DOWN); !ok {
```

Comment:
is there any reason for expecting ISIS session DOWN when README is expecting to session as UP even all members are forwarding-viable False??

Reply:
Looks like README needs to be updated as per https://github.com/openconfig/public/pull/1032/  it is possible that session can be timed out but in that case we may not see neighbor at all so we cant verify it is adjacency-state.  Could you please take that in verification. session state can be DOWN or no session info at all , both cases it should work.

Reply:
Yeah, Darren suggested in Comment above i am using the same approach. After Once ISIS timed out the isis will go down. As per suggestion i updated the same in README as well